gnome-pomodoro 0.24.1 -> 0.26.0

Herschenglime opened the pull request:
## Description of changes
This PR brings gnome-pomodoro up to the latest published version which is compatible with both Gnome 46 and 47. Both the application portion and the extension are functional.

The fix-schema-path.patch had to be changed, as the way that schemas were compiled upstream changed after v0.24 which broke the existing build steps. I also added a preFixup patch to compile these schemas manually.

Fixes NixOS/nixpkgs#314999.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
  - *As far as I'm aware, there were not existing tests for this package and I did not add any. However, it was verified to function locally.*
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Herschenglime committed:
gnome-pomodoro: 0.24.1 -> 0.26.0

Compile schemas manually to follow Nix conventions.

The gnome.post_install(glib_compile_schemas: true) param tries to
compile schemas in `<prefix>/<datadir>/glib-2.0/schemas`, which is not
compatible with the Nix way of handling gsettings schemas.

This commit patches that option out and manually adds a compilation step
to the preFixup hook, which fixes the issue and results in a fully
working local install of the program.

Aleksanaa commented:
Just found out that I'm maintaining three pomodoro packages 😅 please also consider add yourself to maintainers

Bot-wxt1221 reviewed (approved):
LGTM

Herschenglime committed:
maintainers: add herschenglime

This commit adds myself as a maintainer to the gnome-pomodoro package in
addition to adding myself to the overall list.

Herschenglime commented:
> Just found out that I'm maintaining three pomodoro packages 😅 please also consider add yourself to maintainers

Sure, I've added a commit making myself a maintainer on this package. Thanks for your review!

Aleksanaa committed:
gnome-pomodoro: add herschenglime to maintainers
